services/prototype.go

```diff
@@ -595,7 +596,12 @@ func (service *prototype) fetchFileMetadata(ctx context.Context,
 	// FIXME: for now, if a user ORCID is not specified, use the client's ORCID
 	orcid := input.Orcid
 	if orcid == "" {
-		orcid = client.Orcid
+		switch u := userOrClient.(type) {
+		case auth.User:
+			orcid = u.Orcid
+		case auth.Client:
+			orcid = u.Orcid
+		}
```

Comment:
is there some code here that could be refactored?

Reply:
Definitely a matter of taste in my view. Adding a function just to get the field out of a variant struct seems "idiomatic" in object-oriented languages, but to me it's just trading one form of boilerplate for another. This is at least transparent in the sense that it says "if `u` is this type, get this field, and if it's that type, get that one." Also, I think down the line we're probably going to have to have One True Authentication Mechanism for the DTS, and code like this makes it easier to see that we're paying a price for having two.